js/script.js

```diff
@@ -37,40 +37,39 @@ function addProductToHTML(arr) {
    }
 };
 addProductToHTML(productsArr);
-document.querySelectorAll('.btn').forEach((e) => e.addEventListener('click', addBasket));
+document.querySelectorAll('.btn').forEach((e) => e.addEventListener('click', addProductToBasket));
 
 let basketArr = [];
-function addBasket(e) {
-   let idProduct = e.target.getAttribute('id-product');
+function addProductToBasket(e) {
+   const idProduct = e.target.getAttribute('id-product');
    basketArr.push(productsArr[idProduct])
-   update();
+   updateBasketAndCalculate();
 };
 
-function update() {
-   let stringBasket = `Баланс ${balance} грн. <br>`
-   for (const obj of basketArr) {
-      stringBasket += `${obj.names} ${obj.price} грн <br>`
-   }
-   return basketSum(stringBasket);
-};
-
-function basketSum(string) {
+let sumOutside;
+function updateBasketAndCalculate() {
    const doc = document.querySelector('.block-basket');
-   let sum = 0;
-   for (const obj of basketArr) {
-      sum = sum + obj.price
-   }
-   string += `Итого:${sum} грн <button class="buy">Оформить заказ</button>`;
-   doc.innerHTML = string;
-   document.querySelector('.buy').onclick = aproveOrder;
+   let stringBasket = `Баланс ${balance} грн. <br>`;
 
-   function aproveOrder() {
-      let block = document.querySelector('.block-basket');
-      basketArr = [];
-      balance = balance - sum;
-      block.innerHTML = `Заказ оформлен. Сумма баланса ${balance} грн`;
-   };
-};
+   // for (const obj of basketArr) {
+   //    sum = sum + obj.price
+   //    stringBasket += `${obj.names} ${obj.price} грн <br>`
+   // };
 
+   const sum = basketArr.reduce((total, obj) => total + obj.price, 0)
+   stringBasket += basketArr.reduce((total, obj) => total += `${obj.names} ${obj.price} грн <br>`, '')
```

Comment:
you can use object for reduce method and accumulate both sum & stringBasket inside object, so you would not use reduce twice